SECOND TASK> follow the same process from the DM2024-Lab1-Master on the new dataset. You don't need to explain all details as we did (some minimal comments explaining your code are useful though). This part is worth 30% of your grade.
Download the the new dataset. The dataset contains a sentiment and comment columns, with the sentiment labels being: 'nostalgia' and 'not nostalgia'. Read the specificiations of the dataset for background details.
You are allowed to use and modify the helper functions in the folder of the first lab session (notice they may need modification) or create your own.


1. Check the basic information of the dataset

In [9]:
import pandas as pd

file_path = r'D:\Visual Studio Code\111\td\Nostalgic_Sentiment_Analysis_of_YouTube_Comments_Data.csv'
df = pd.read_csv(file_path)

# View the first few rows of the dataset
print(df.head())

# View the column names of the dataset
print("Column names:", df.columns)

# View the data types of the dataset
print("Data types:\n", df.dtypes)


       sentiment                                            comment
0  not nostalgia  He was a singer with a golden voice that I lov...
1      nostalgia  The mist beautiful voice ever I listened to hi...
2      nostalgia  I have most of Mr. Reeves songs.  Always love ...
3  not nostalgia  30 day leave from 1st tour in Viet Nam to conv...
4      nostalgia  listening to his songs reminds me of my mum wh...
列名： Index(['sentiment', 'comment'], dtype='object')
資料類型：
 sentiment    object
comment      object
dtype: object


Answer:

1.	評論數據（comment）：資料集包含一系列評論，這些評論是人們對於某個主題的看法或感受。每一條評論都與一個情感標籤相關聯，表明這條評論是否表達了懷舊情緒。
例如：
第0條評論是“not nostalgia”，內容是：“He was a singer with a golden voice that I love to hear all the time...”，意味著這條評論沒有表達懷舊情緒。
第1條評論是“nostalgia”，內容是：“The mist beautiful voice ever I listened to him when I was a kid...”，表明這條評論表達了懷舊情緒。
2.	情感標籤（sentiment）：每條評論都伴隨一個情感標籤，這個標籤是“nostalgia”（懷舊）或“not nostalgia”（不懷舊）。這個標籤用於分類評論，指示評論者在撰寫評論時是否體驗到了懷舊情緒。
3.	列名和資料類型：資料集有兩列，分別是情感（sentiment）和評論（comment）。
資料類型為object，通常表示這些列包含的是文本資料（字串類型），而非數值資料。


2. Check the detailed description of the dataset

In [7]:
# Check the basic info of the dataset, including the number of non-null values for each column
print("Dataset basic info:\n", df.info())
# Check the number of unique values for each column
print("\nNumber of unique values per column:\n", df.nunique())
# Check the shape of the dataset, i.e., the number of rows and columns
print("\nShape of the dataset:", df.shape)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1500 entries, 0 to 1499
Data columns (total 2 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   sentiment  1500 non-null   object
 1   comment    1500 non-null   object
dtypes: object(2)
memory usage: 23.6+ KB
資料集基本資訊：
 None

每列的唯一值數量：
 sentiment       2
comment      1499
dtype: int64

資料集的形狀： (1500, 2)


Answer:

下面是對輸出資訊的逐項解釋：
1.	DataFrame 類型和大小：RangeIndex: 1500 entries, 0 to 1499 說明這個 `DataFrame` 包含1500行資料，索引從0到1499。
2.	數據列：DataFrame 總共有2列。0 sentiment 1500 non-null object` 表示 sentiment 列有1500個非空值，資料類型為object（通常是字串）。1 comment 1500 non-null object 表示 comment 列也有1500個非空值，資料類型同樣為 object。
3.	記憶體使用：這個 DataFrame 在記憶體中佔用的大小超過23.6千位元組。
4.	每列的唯一值數量： sentiment 列有2個唯一值，這兩個值是“懷舊”和“不懷舊”。comment 列有1499個唯一值，這意味著有1499條不同的評論，或者有極少數評論內容重複。
5.	資料集的形狀：資料集的形狀:(1500,2) 再次確認了這個 DataFrame 有1500行和2列。
綜上所述，這個輸出提供了一個關於資料集的全面快照，包括它的結構、資料類型、記憶體使用情況、每列的唯一值數量以及資料集的維度。這些資訊對於理解資料集的特性和進行初步的資料探索非常有用。


3. Check the missing values in the dataset

In [10]:
# Check the number of missing values for each column
print("\nNumber of missing values per column:\n", df.isnull().sum())
# Check the total number of missing values in the dataset
print("\nTotal number of missing values:", df.isnull().sum().sum())


每列的缺失值數量：
 sentiment    0
comment      0
dtype: int64

總缺失值數量： 0


Answer:

這段輸出資訊提供了關於資料集中缺失值的統計資料。具體來說：
1.	每列的缺失值數量：在 sentiment 這一列中，沒有缺失值。這意味著所有1500條資料都有對應的情感標籤。
2.	資料類型：表明這個由缺失值統計構成的 Series 的資料類型是 int64，即每個元素都是一個整數。
3.	總缺失值數量：總缺失值數量:0 表示在整個資料集中，沒有任何缺失值。這是一個非常理想的狀態，意味著所有的資料行都是完整的，沒有需要填充或處理的缺失資料。
綜上所述，這個輸出告訴我們，這個資料集在 `sentiment` 列中沒有缺失值，而且整個資料集也沒有缺失值，這使得資料分析和處理變得更加直接和簡單，不需要進行額外的資料清洗步驟來處理缺失值。不過，關於 `comment` 列的資料類型，可能需要進一步確認，以確保資料類型的準確性。
